### ISSUE TITLE
false nonlocal binding error for `__class__`

### ISSUE THREAD

**Issue by youknowone** [RustPython/RustPython#2008]
## Feature

```python
class A:
    def f(self):
        nonlocal __class__
        __class__ = int
```

This is legal code in Python but raising error in RustPython:
```
SyntaxError: no binding for nonlocal '__class__' found at line 0 column 0
```

**Comment by BolunThompson** [comment]
How are you getting that? An error is being raised for me in Python.

```python
Python 3.8.3 (default, May 17 2020, 18:15:42)
[GCC 10.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def f():
...     nonlocal x
...     x = 10
...
  File "<stdin>", line 2
SyntaxError: no binding for nonlocal 'x' found
```

**Comment by youknowone** [comment]
I am sorry, I revised the code.

**Comment by youknowone** [comment]
I think this is solved by  #2190
